spot_ros2_control/launch/spot_ros2_control.launch.py

```diff
@@ -14,11 +18,82 @@
 
 from spot_driver.launch.spot_launch_helpers import get_login_parameters, spot_has_arm
 
+THIS_PACKAGE = "spot_ros2_control"
+
+
+def create_controllers_config(spot_name: str, has_arm: bool) -> None:
+    """Writes a configuration file used to put the ros2 control nodes into a namespace.
+    This is necessary as if your ros2 control nodes are launched in a namespace, the configuration yaml used
+    must also reflect this same namespace when defining parameters of your controllers.
+
+    Args:
+        spot_name (str): Name of spot. If it's the empty string, the default controller file with no namespace is used.
+        has_arm (bool): Whether or not your robot has an arm. Necessary for defining the joints that the forward
+                        position controller should use.
+
+    Returns:
+        str: Path to controllers config file to use
+    """
+
+    arm_text = "with_arm" if has_arm else "without_arm"
+    template_filename = os.path.join(
+        get_package_share_directory(THIS_PACKAGE), "config", f"spot_default_controllers_{arm_text}.yaml"
+    )
+
+    if spot_name:
+        with open(template_filename, "r") as template_file:
+            config = yaml.safe_load(template_file)
+            forward_position_controller_joints = config["forward_position_controller"]["ros__parameters"]["joints"]
+            config["forward_position_controller"]["ros__parameters"]["joints"] = [
+                f"{spot_name}/{joint}" for joint in forward_position_controller_joints
+            ]
+            config[f"{spot_name}/controller_manager"] = config["controller_manager"]
+            del config["controller_manager"]
+            config[f"{spot_name}/forward_position_controller"] = config["forward_position_controller"]
+            del config["forward_position_controller"]
+        with NamedTemporaryFile(suffix=".yaml", mode="w", delete=False) as out_file:
+            yaml.dump(config, out_file)
+            return out_file.name
```

Comment:
Was unsure how to handle the cleanup of the temporary file -- currently they are not deleted, otherwise they would go out of scope when this function exits and be deleted before launch can actually use them 
@mhidalgo-bdai

Reply:
It's fine. [`launch_ros`](https://github.com/ros2/launch_ros/blob/38152b2d04c6b6f126d31726ac9e12888b2e2e1f/launch_ros/launch_ros/actions/node.py#L369C9-L377) doesn't do any cleanup either. OS will drop them on reboot.

Problem here is `launch` not exposing good mechanisms for resource management. You could use a shutdown even handler to deal with it but it's messy.